source/io/module_javaScriptInvoke.js

```diff
@@ -452,32 +392,28 @@ var assignJavaScriptInvoke;
 
         Module.javaScriptInvoke.jsJavaScriptInvoke = function (
             occurrencePointer,
-            functionNamePointer,
+            functionNameTypeRef,
+            functionNameDataRef,
             returnPointer,
             parametersPointer,
             parametersCount,
             isInternalFunction,
             errorTypeRef,
             errorDataRef) {
             var errorValueRef = Module.eggShell.createValueRef(errorTypeRef, errorDataRef);
+            var functionNameValueRef = Module.eggShell.createValueRef(functionNameTypeRef, functionNameDataRef);
+            var functionName = Module.eggShell.readString(functionNameValueRef);
+            var parameters = [];
 
-            var newError = {
-                status: false,
-                code: ERRORS.NO_ERROR.CODE,
-                source: ERRORS.NO_ERROR.MESSAGE
-            };
+            var returnValueRef = createValueRefFromPointerArray(returnPointer, 0);
+            if (isInternalFunction) {
+                parameters = parameters.concat(returnValueRef);
```

Comment:
should switch to `parameters.push(returnValueRef)` to avoid extra array allocations like we mentioned in the book club :)